### PR TITLE
Blocks

### PR DESCRIPTION
**Pull request by klementc**
New feature: trigger blocks
It doesn't break any of the previous levels but adds some new possibilities.

Trigger blocs can be used for different purposes depending on their "name":
- SKN: set random skin to player
- BEM: create beam
- SHW/DEL: set similarly to TPs (SHWa1 paired with SHWa2) but to create blocks that appear/disappear upon touching their paired trigger
- POP: to create blocks that pop on / off
- JMP: jumpy blocks
- COL: change background color
- INF/NOF/NRM: change flow tag to dynamically change the flow type inside a level
- STF: set the number of flow points of the player

Nothing should break on older levels, and the level designers can create them now. Demo videos of the trigger blocks by xakaze: https://www.youtube.com/watch?v=Rt2uvxCtZKE  

In our tests, it doesn't add more lag than other blocks

### CODE REVIEW

File: game/classes/Player.ts [hunk 1]
```diff
@@ -136,14 +136,90 @@ export class Player extends TileObject {
...
+          this.handType = Math.floor(Math.random() * 24);
+          this.colour = Math.floor(Math.random() * 0xFFFFFF);
+          this.colour2 = Math.floor(Math.random() * 0xFFFFFF);
+          game.addSkin();
+        } 
+        /* trigger to emit beams */
+        else if (trigger.typeTrigger.includes("BEM")) {
+          (this.parent as Game).emitBeam(this.x,this.y,this.width);
+        } else if (trigger.typeTrigger.includes("COL")) {
+          this.curLevel.colorBG = trigger.typeTrigger.slice(3) == "RND" ? Math.random()*0xFFFFFFFF : +trigger.typeTrigger.slice(3);
```
Comment (kiootic) [inline game/classes/Player.ts:201]: Treating it as hex color would be easier to use:
```suggestion
          this.curLevel.colorBG = trigger.typeTrigger.slice(3) == "RND" ? Math.random()*0xFFFFFFFF : (parseInt(trigger.typeTrigger.slice(3),16)||0);
```

File: game/classes/Game.ts [hunk 2]
```diff
@@ -1053,6 +1053,7 @@ export class Game extends lib.flash.display.MovieClip {
   }
 
   public ping(e: lib.flash.events.Event = null): any {
+    Anim.colourMe(this.skyLine, this.level.colorBG);
```
Comment (kiootic) [inline game/classes/Game.ts:1056]: This make singleplayer BG gone.

```suggestion
    if (this.level.colorBG != null) {
      Anim.colourMe(this.skyLine, this.level.colorBG);
    }
```

**Review by kiootic — CHANGES_REQUESTED** [review]
I'd like some cleanup on the `console.log` to avoid spamming console.

**Comment by klementc** [comment]
Console spam should be fixed now (thx for additional bg color check)

**Pull request merged**